image/lib.sh

```diff
@@ -28,20 +35,78 @@ function errnullify     { "$@" 2>/dev/null; }
 ## Utils
 ##
 
+abspath() {
```

Comment:
Added a few utils to lib, as they're generic, useful, and I'd rather stop duplicating them in scripts.

Reply:
I'll migrate `is-newer-than` to be it's own standalone in `image/bin` in a follow up, as looking at it now, it doesn't have any need to be ran in-process like the others do.

Reply:
Update: went ahead and preemptively migrated `is-newer-than` in a commit here.